### PR TITLE
feat(argocd-apps): multiple sources for Application and ApplicationSet

### PR DESCRIPTION
**Pull request by toyamagu-2021**
## Description

* Add `spec.sources` for application and applicationsets.

## Manual tests

* Single source Application
  *  `helm upgrade -i argocd-apps-applications charts/argocd-apps -n argocd --values charts/argocd-apps/ci/applications-values.yaml`
* Multiple sources Application
  *  `helm upgrade -i argocd-apps-applications-ms charts/argocd-apps -n argocd --values charts/argocd-apps/ci/applications-multiple-sources-values.yaml`
* Single source ApplicationSets
  *  `helm upgrade -i argocd-apps-applicationsets charts/argocd-apps -n argocd --values charts/argocd-apps/ci/applicationsets-values.yaml`
* Multiple sources ApplicationSets
  *  `helm upgrade -i argocd-apps-applicationsets-ms charts/argocd-apps -n argocd --values charts/argocd-apps/ci/applicationsets-multiple-sources-values.yaml`

Signed-off-by: toyamagu <toyamagu2021@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

**Review by yu-croco — APPROVED** [review]
Thank you for your contribution!
LGTM.

File: charts/argocd-apps/ci/applicationsets-values.yaml [hunk 1]
```diff
@@ -1,8 +1,7 @@
 # Test with applicationsets
 
 applicationsets:
-- name: guestbook
-  namespace: default # Only for test purpose.
+- name: applicatioset
```
Comment (pdrastil) [inline charts/argocd-apps/ci/applicationsets-values.yaml:4]: Small typo here - why this was changed from Argo sample app?

**Review by pdrastil — APPROVED** [review]
LGTM

File: charts/argocd-apps/ci/applicationsets-values.yaml (hunk 1 above)
Comment (toyamagu-2021) [reply charts/argocd-apps/ci/applicationsets-values.yaml]: Thank you. Fixed it!
Just to avoid a name collision.

**Pull request merged**